cockpit/templates/api/api-configmap.yaml

```diff
@@ -191,31 +191,24 @@ data:
     auth:
       callbackUrl: {{ .Values.api.ingress.path }}/auth/login/callback
 
-      {{- if and .Values.authentication.github .Values.authentication.github.clientId }}
+      {{- if .Values.authentication.github }}
       github:
-        userAutorizationUri: https://github.com/login/oauth/authorize
-        accessTokenUri: https://github.com/login/oauth/access_token
-        userProfileUri: https://api.github.com/user
-        codeParameter: code
-        responseType: code
-        clientId: {{ .Values.authentication.github.clientId }}
-        clientSecret: {{ .Values.authentication.github.clientSecret }}
         redirectUri: https://{{ index .Values.api.ingress.hosts 0 }}{{ .Values.api.ingress.path }}/auth/login/callback?provider=github
-        scopes:
+{{ toYaml .Values.authentication.github | indent 8 }}
       {{- end }}
 
-      {{- if and .Values.authentication.google .Values.authentication.google.clientId }}
+      {{- if .Values.authentication.google }}
       google:
-        userAutorizationUri: https://accounts.google.com/o/oauth2/v2/auth
-        accessTokenUri: https://oauth2.googleapis.com/token
-        userProfileUri: https://openidconnect.googleapis.com/v1/userinfo
-        codeParameter: code
-        responseType: code
-        clientId: {{ .Values.authentication.google.clientId }}
-        clientSecret: {{ .Values.authentication.google.clientSecret }}
         redirectUri: https://{{ index .Values.api.ingress.hosts 0 }}{{ .Values.api.ingress.path }}/auth/login/callback?provider=google
-        scopes: openid,profile,email
+{{ toYaml .Values.authentication.google | indent 8 }}
       {{- end }}
+
+      {{- if .Values.authentication.oidc }}
+      oidc:
+        redirectUri: https://{{ index .Values.api.ingress.hosts 0 }}{{ .Values.api.ingress.path }}/auth/login/callback?provider=oidc
```

Comment:
IMHO, it's better to allow the user to override the `redirectUri` by himself.
So I would propose to check if `.Values.authentication.oidc.redirectUri` is defined. If it is, use it, otherwise, calculate it.